lib/meilisearch/http_request.rb

```diff
@@ -17,6 +17,10 @@ def initialize(url, api_key = nil)
     end
 
     def http_get(path = '', query = {})
+      # Flattens arguments as array for search params
+      query.each do |key, value|
+        query[key] = value.join(',') if value.is_a?(Array)
+      end
       response = self.class.get(
         @base_url + path,
         query: query,
```

Comment:
At this line, I would rather add:
```ruby
query: query.map { |_, v| v.is_a?(Array)? v.join(',') : v }
```

Or, to avoid the one line, instead of what you wrote you can write:
```ruby
# Flattens arguments as array for search params
parsed_query = query.map do |_, v|
  v.is_a?(Array)? v.join(',') : v
end
```
and in the `class.get` method:
```ruby
query: parsed_query,
```
(you can find a better name for `parsed_query` of course)

In ruby, we would rather use the functional way = recreate variable instead of modifying the first one. No need to use the `key` in the loop with this solution 😉

Reply:
(Or less aggressive:

```ruby
# Flattens arguments as array for search params
parsed_query = query.map do |k, v|
  new_value = v.is_a?(Array)? v.join(',') : v
  [k, new_value]
end.to_h
```
😇 )